update_branch_protections.py

```diff
@@ -0,0 +1,234 @@
+from __future__ import annotations
+
+import argparse
+import csv
+from dataclasses import dataclass, fields
+from pathlib import Path
+from typing import List
+
+from update_github_settings import (BranchProtection, Repository,
+                                    default_required_status_checks)
+
+master_protect = BranchProtection()
+gh_pages_prot = BranchProtection(pattern='gh-pages', allows_force_pushes=True,
+                                 dismisses_stale_reviews=False,
+                                 required_status_checks=[],
+                                 requires_status_checks=False,
+                                 required_approving_review_count=0,
+                                 requires_approving_reviews=False,
+                                 )
+all_prot = BranchProtection(pattern='*', required_status_checks=[],
+                            dismisses_stale_reviews=False,
+                            requires_status_checks=False,
+                            required_approving_review_count=0,
+                            requires_approving_reviews=False,
+                            allows_deletions=True,)
+
+
+def str_to_bool(val: str):
+    return val.lower() in ('y', 'yes', 'true')
+
+
+@dataclass
+class ProtectionGroup:
+    owner: str
+    repo_name: str
+    repo_type: str
+    master: bool = False
+    gh_pages: bool = False
+    default: bool = False
+
+    # Map column names to protection rules
+    RULE_MAP = {
+        'master': master_protect,
+        'gh_pages': gh_pages_prot,
+        'default': all_prot
+    }
+
+    # map repo-types to status-check lists
```

Comment:
These refer to the [check-lists here](https://github.com/pcdshub/pcds-python-migration-tools/blob/660df8729c601d5da95b8212481053a5897c5e8c/update_github_settings.py#L22-L42)